nautobot/docs/user-guide/administration/installation/health-checks.md

```diff
@@ -0,0 +1,307 @@
+
+- the server is running
+- and the server can connect to the database
+- and all Django migrations have been applied
+- and the server can connect to Redis
+- and the server can write to an appropriate location on the filesystem
+- and the server is not too busy handling other requests to respond to this request.
+
+Similarly, but not identically, the CLI command `nautobot-server health_check` should run and return an exit code of `0` (success) so long as:
+
+- the command can connect to the database
+- and all Django migrations have been applied
+- and the command can connect to Redis
+- and the command can write to an appropriate location on the filesystem
+
+Note the differences between these two. In some situations you'll want to use both for different types of checks. More on this later in this document.
+
+### Nautobot Celery Worker
+
+In addition to monitoring the existence of a given Celery worker process ID, you can use the fact that Celery provides a [`celery inspect ping` CLI command](https://docs.celeryq.dev/en/stable/reference/cli.html#celery-inspect) that sends a short message to a given Celery worker(s) and reports back on whether it receives a response(s). Nautobot wraps this with the `nautobot-server` CLI command, so in general you can run `nautobot-server celery inspect ping --destination <worker name>` to confirm whether a given worker is able to receive and respond to Celery control messages.
+
+### Nautobot Celery Beat
+
+In addition to monitoring the Celery Beat process ID, you can use the fact that Nautobot's custom Celery Beat scheduler respects the [`CELERY_BEAT_HEARTBEAT_FILE`](../configuration/optional-settings.md#celery_beat_heartbeat_file) configuration setting, which specifies a filesystem path that will be repeatedly `touch`ed to update its last-modified timestamp so long as the scheduler is running. You can check this timestamp against the current system time to detect whether the Celery Beat scheduler is firing as expected. One way is using the `find` command with it's `-mmin` parameter, and checking whether it finds the expected file with a recent enough modification time (here, 0.1 minutes, or 6 seconds) or not:
+
+```shell
+[ $(find $NAUTOBOT_CELERY_BEAT_HEARTBEAT_FILE -mmin -0.1 | wc -l) -eq 1 ] || false
+```
+
+### Databases
+
+#### PostgreSQL
+
+PostgreSQL provides the [`pg_isready` CLI command](https://www.postgresql.org/docs/current/app-pg-isready.html) to check whether the database server is running and accepting connections.
+
+#### MySQL
+
+While MySQL provides the [`mysqladmin ping` CLI command](https://dev.mysql.com/doc/refman/8.0/en/mysqladmin.html), it's important to note that this command only checks whether the database server is running - it still exits with return code `0` if the server is running but not accepting connections. Therefore you might in some cases wish to run a command that actually connects to the database, such as `mysql --execute "SHOW DATABASES;`.
```

Comment:
Should we **just** recommend `mysql --execute "SHOW DATABASES;` to keep it consistent with PostgreSQL?

Reply:
My thinking was that `mysqladmin ping` would be applicable for startup and liveness probes, while `mysql --execute "SHOW DATABASES"` is applicable for readiness probes. Open to changing that though.